transport_analysis/conductivity_msd.py

```diff
@@ -0,0 +1,474 @@
+from MDAnalysis.core.groups import UpdatingAtomGroup
+from MDAnalysis.exceptions import NoDataError
+from MDAnalysis.units import constants
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats
+from transport_analysis.utils import (
+    msd_fft_cross_1d,
+    msd_variance_cross_1d,
+    average_directions,
+)
+
+if TYPE_CHECKING:
+    from MDAnalysis.core.universe import AtomGroup
+
+
+## Need to pass the entire the universe to get the com of the universe not just the atomg
+class ConductivityEinstein(AnalysisBase):
+    """
+        Class to calculate the diffusion_coefficient of a species.
+        Note that the slope of the mean square displacement provides
+        the diffusion coeffiecient. This implementation uses FFT to compute MSD faster
+        as explained here: https://stackoverflow.com/questions/34222272/computing-mean-square-displacement-using-python-and-fft
+        Since we are using FFT, the sampling frequency (or) dump frequency of the simulation trajectory
+        plays a critical role in the accuracy of the result.
+
+        Particle velocities are required to calculate the viscosity function. Thus
+        you are limited to MDA trajectories that contain velocity information, e.g.
+        GROMACS .trr files, H5MD files, etc. See the MDAnalysis documentation:
+        https://docs.mdanalysis.org/stable/documentation_pages/coordinates/init.html#writers.
+
+        Parameters
+        ----------
+    atomgroup : AtomGroup
+        An MDAnalysis :class:`~MDAnalysis.core.groups.AtomGroup`.
+        Note that :class:`UpdatingAtomGroup` instances are not accepted.
+    temp_avg : float (optional, default 300)
+        Average temperature over the course of the simulation, in Kelvin.
+    dim_type : {'xyz', 'xy', 'yz', 'xz', 'x', 'y', 'z'}
+        Desired dimensions to be included in the viscosity calculation.
+        Defaults to 'xyz'.
+    linear_fit_window : tuple of int (optional)
+        A tuple of two integers specifying the start and end lag-time for
+        the linear fit of the viscosity function. If not provided, the
+        linear fit is not performed and viscosity is not calculated.
+
+    Attributes
+    ----------
+    atomgroup : :class:`~MDAnalysis.core.groups.AtomGroup`
+        The atoms to which this analysis is applied
+    dim_fac : int
+        Dimensionality :math:`d` of the viscosity computation.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged viscosity function over all the particles with respect
+        to lag-time. Obtained after calling :meth:`ViscosityHelfand.run`
+    results.visc_by_particle : :class:`numpy.ndarray`
+        The viscosity function of each individual particle with respect to
+        lag-time.
+    results.viscosity : float
+        The viscosity coefficient of the solvent. The argument
+        `linear_fit_window` must be provided to calculate this to
+        avoid misinterpretation of the viscosity function.
```

Comment:
I don't think these are correct for this class, the parameters in results should be documented here.